### ISSUE TITLE
[Test]Sometimes test for test_silvermoon_portal() fails

### ISSUE THREAD

**Issue by tonyyyye** [jleclanche/fireplace#443]
When the ("KAR_077") silvermoon portal summons dire_wolf_alpha the attack value of the designated minion will not be as expected.

**Comment by tonyyyye** [comment]
>       assert whelp.atk == 3
E    AssertionError: assert 4 == 3
E     +  where 4 = <Minion ('Whelp')>.atk

This is how my case fails

**Comment by tonyyyye** [comment]
assert whelp.atk == 3
E AssertionError: assert 4 == 3
E + where 4 = <Minion ('Whelp')>.atk

This is how my case fails 


Probably some related cases may fail due to the same reason.
Will check when go through cases.

**Comment by jleclanche** [comment]
Yeah I'm aware of the failure mode. Maybe you can remove the other minion from the board after you've checked for it, I think there's a way to do it that doesn't trigger deathrattles (remove_from_game() was it?)

**Comment by tonyyyye** [comment]
Currently I haven't found remove_from_game().
All related operations are implemented as set:
  target.zone = Zone.GRAVEYARD

And I think it is a good idea to add this to Minion and Weapon.

**Comment by jleclanche** [comment]
@tonyyyye Make sure you're on Discord at https://discord.gg/hearthsim-devs -- We can chat there more easily and you can ask anything you like :)

**Comment by tonyyyye** [comment]
Besides Dire_wolf_alpha, Flametongue_Totem, Wild_Pyromancer will also cause the problem.
Wile all minion related could be solved by game.current_player.field[-1].bounce(), 
Wild_Pyromancer is another problem, which was caused by modification of spell cycle in patch 9.2 .

> Hearthstone Mechanics Update – We’ve changed some underlying rules to prevent Wild Pyromancer from triggering off of Mindgames

**Comment by tonyyyye** [comment]
There are no more after-spell-damages when summon a Wild Pyromancer, including this silvermoon portal.
The rest of the excetion cases are tested with like a 10k times without error, so I think it can be modified by that way.

The Wild Pyromancer("NEW1_020")  problem will be mentioned by another issue.
